zerver/lib/events.py

```diff
@@ -339,7 +339,10 @@ def our_person(p):
                 # Convert the emails to user_profile IDs since that's what register() returns
                 # TODO: Clean up this situation by making the event also have IDs
                 for item in event["subscriptions"]:
-                    item["subscribers"] = [get_user_profile_by_email(email).id for email in item["subscribers"]]
+                    item["subscribers"] = [
+                        get_user(email, user_profile.realm).id
```

Comment:
This will work only if the subscribers are in the same context. Should it be filtered?